### PR TITLE
Fixed manual searching for scene named anime shows.

### PR DESCRIPTION
**Pull request by p0psicles**
Somehow results = curProvider.cache.update_cache_manual_search(searchResults[curEp]) was always running with curEp= -1. As it should run the function with the episode number as a param.

With this fix it was getting all results for show Fairy Tails S2, for providers nyaatorrent and tokyotosho. As before it only found results from nzb providers.

We'll need to thoroughly test this will all different kinds of shows. I can also image that for some shows, you will get allot more results now.
@fernandog @labrys @duramato

### CODE REVIEW

File: sickbeard/search.py [hunk 1]
```diff
@@ -565,7 +565,10 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
 
         # Update the cache if a manual search is being runned
         if manual_snatch:
-            results = curProvider.cache.update_cache_manual_search(searchResults[curEp])
+            results = False
```
Comment (labrys) [inline sickbeard/search.py]: Not needed

File: sickbeard/search.py [hunk 2]
```diff
@@ -565,7 +565,10 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
...
+            for searched_episode in episodes:
+                results = curProvider.cache.update_cache_manual_search(searchResults[searched_episode.episode])
```
Comment (labrys) [inline sickbeard/search.py]: Should be a comprehension:
```python
results = [curProvider.cache.update_cache_manual_search(searchResults[searched_episode.episode]) for searched_episode in episodes]
```

File: sickbeard/search.py [hunk 3]
```diff
@@ -565,7 +565,10 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
...
+
             if results:
                 # If we have at least a result from one provider, it's good enough to be marked as result
                 finalResults.append(results)
```
Comment (labrys) [inline sickbeard/search.py]: The existing code only appends the result for the last found.
With the change of results to a list comprehension this should become a list concat.  In addition the concat should be moved outside the `if` statement which will only be used to continue.

Comment (p0psicles) [inline sickbeard/search.py]: As discussed we're collecting a list of bools here? With what purpose? In line 732 is picked up again with finalResults += set(multiResults.values()). So then results are added to it?

I'll look further into this, if we can agree the solution for the caching is the correct one. I haven't really looked into the finalResults, so will do that first.

**Comment by p0psicles** [comment]
@labrys can you approve this?

**Comment by fernandog** [comment]
@p0psicles 
I gave one of the approves

File: sickbeard/search.py [hunk 4]
```diff
@@ -470,6 +470,7 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
     """
     foundResults = {}
     finalResults = []
+    manual_snatch_results = []
```
Comment (fernandog) [inline sickbeard/search.py]: @p0psicles I remember I added this. maybe in one of the rebases it got lost

**Pull request merged**